### PR TITLE
Disable "automatically set focus to focusable elements" by default

### PR DESCRIPTION
**Pull request by codeofdusk**
### Link to issue number:
None.

### Summary of the issue:
Especially after #11130, it may be worth considering disabling "automatically set system focus to focusable elements" by default.

### Description of how this pull request fixes the issue:
* Disables the option by default in the config spec.
* Moves the checkbox to the browse mode panel (with the other virtual buffers options).
* Updates the user guide.

### Testing performed:
Tested that toggling the checkbox works as intended, and that NVDA+8 still works.

### Known issues with pull request:
None.

### Change log entry:
=== Changes ===
- NVDA no longer automatically sets the system focus to focusable elements by default in browse mode, improving performance and stability. (#11190)

### CODE REVIEW

**Comment by codeofdusk** [comment]
I've been using fast browse mode on my system for a while since #11130 was merged, but leaving as a draft until I know it's stable enough/there are no known major issues.

Cc @jcsteh, @leonardder, @mltony, @michaelDCurran.

**Comment by Adriani90** [comment]
At least from an user perspective I can tell that it's working very good now, so it could be enabled by default. This needs to be documented properly in the changes.t2t, in the userguide and maybe also in the developers guide (i.e. shift+tab will still report the element with system focus, switching to focus mode and pressing tab shows a different behavior than pressing tab in browse mode because system focus will move from its last position which is different from the virtual cursor, etc.). A good documentation is very crucial here because otherwise we will get many bug reports by accessibility testers regarding NVDA not scrolling the screen, pressing tab not doing what is expected to do etc.
And elieve me, to provide the proper information by commenting on every bug and saying this is expected behavior can cost alot of time and can be quite frustrating after some time.

**Comment by jcsteh** [comment]
There's one major outstanding issue which I think needs to be fixed before
this is enabled by default. If you force focus mode with NVDA+space, focus
is not routed to the focusable object under the cursor. IMO, that's an
unacceptable regression for default behaviour. I started looking into this,
but it's not quite as trivial to fix as I originally thought.

**Comment by codeofdusk** [comment]
PR no longer blocked, since #11206 was merged.

File: source/config/configSpec.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # -*- coding: UTF-8 -*-
-#A part of NonVisual Desktop Access (NVDA)
-#Copyright (C) 2006-2019 NV Access Limited, Babbage B.V., Davy Kager, Bill Dengler
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2006-2010 NV Access Limited, Babbage B.V., Davy Kager, Bill Dengler
```
Comment (feerrenrut) [inline source/config/configSpec.py:3]: I think this should be 2020

**Comment by feerrenrut** [comment]
I'd like this to have a longer run in alpha, I'll look to review / merge this after branching beta.

**Comment by codeofdusk** [comment]
Now that `master` has moved onto 2020.3, I think this is worth merging. This should be well tested for as long as possible.

File: source/gui/settingsDialogs.py [hunk 2]
```diff
@@ -1913,6 +1913,16 @@ def makeSettings(self, settingsSizer):
 		self.trapNonCommandGesturesCheckBox = sHelper.addItem(wx.CheckBox(self, label=trapNonCommandGesturesText))
 		self.trapNonCommandGesturesCheckBox.SetValue(config.conf["virtualBuffers"]["trapNonCommandGestures"])
 
+		# Translators: This is the label for a checkbox in the
+		# browse mode settings panel.
+		autoFocusFocusableElementsText = _("Automatically set system &focus to focusable elements")
+		self.autoFocusFocusableElementsCheckBox = sHelper.addItem(
+			wx.CheckBox(self, wx.ID_ANY, label=autoFocusFocusableElementsText)
```
Comment (feerrenrut) [inline source/gui/settingsDialogs.py:1920]: The use of `wx.ID_ANY` is not required.
```suggestion
			wx.CheckBox(self, label=autoFocusFocusableElementsText)
```

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Happy to merge this, one minor issue that I missed previously.

**Review by feerrenrut — APPROVED** [review]
Thanks @codeofdusk

**Comment by Adriani90** [comment]
@feerrenrut, @jcsteh, now that this has become the Default behavior, what should we do now with issues like #2039, #5251, #5831, #6282, #6678 or #8999?

#2039 and #8999 describe similar behavior, however I let both open because both contain valuable Input.
Suggestions are saying that this is related to bad web authoring, but it could also be a Change in Rendering time between NVDA 2018.3 and 2018.4 which led to this behavior of Focus jumping.
Having in mind that this PR solves the issues referenced in this comment, is there a reason to Keep them open still? Will there be any Investigation to further improve this even with automatic set Focus to focusable Elements turned on?

**Comment by Adriani90** [comment]
Ah there are also issues #8079 and #8010. These would also be fixed when using the default behavior provided here, but not when the focus is set automatically to focusable elements.

**Comment by codeofdusk** [comment]
Since fast browse mode is now the default and users are warned that disabling it may have performance and stability implications, I see no need to fix these issues when fast browse mode is off.

**Comment by jage9** [comment]
One of the main reasons for keeping this on for me was to use the mouse routing command to click on elements that could not be accessed otherwise. How will the new browse mode effect this? Should there be/is there a command to temporarily move focus, or is it a part of the move mouse to navigator object command? I turned the new option off since this is the recommended behavior, so will report any results.

**Comment by Adriani90** [comment]
How about low vision users who need also the scrolling of the screen? Or does the screen scroll automatically though?


Von meinem iPhone gesendet

**Comment by lukaszgo1** [comment]
Having this enabled by default is quite problematic for people who have disabled "caret moves review cursor" as there is no reliable way to move navigator object to the position of the browse mode  caret. Being able to move it there is critical as we can only route mouse to the navigator not to the caret. I've created #9622 some time ago and I believe that if it would be fixed this situation would dramatically improve. @feerrenrut I understand that we are in the features freeze and #9622 can be technically classified as a feature request / enhancement but without it the new behavior is quite problematic for some people as explained above. Would you accept a pr fixing it i.e moving navigator object to focus moves navigator to the position of the browse mode caret?

**Comment by lukaszgo1** [comment]
@feerrenrut Have you seen [this commend](https://github.com/nvaccess/nvda/pull/11190#issuecomment-649819536) of mine? Could you please share your opinion? In the current state people who are  more comfortable working with review cursor not following caret are forced to enable "set focus to focusable elements" to be able to work effectively on webpages requiring mouse/object navigation.

**Comment by feerrenrut** [comment]
@lukaszgo1 Could you please create a new issue to explain this issue fully? Or are you saying that #9622 already covers the situation? If so, please update #9622 to explain how this PR fits in. 

> Would you accept a pr fixing it i.e moving navigator object to focus moves navigator to the position of the browse mode caret?

I'm not really sure that I understand this.

**Comment by lukaszgo1** [comment]
I hope #9622 after my recent edit clears things up

**Comment by Adriani90** [comment]
since #9622 has been fixed, I think there is no other issue open on this. Given that the screen scrolls automatically and the focus is moved to the virtual cursor when swithicng to focus mode automatically or with nvda+space bar, I think it makes sense to close those issues dealing with focus jumping etc. I will proceed with that.

**Comment by ntcarver0** [comment]
If testing confirms how awesome this improvement is (I'm seeing quite the positive difference already), would it be possible to automatically enable this feature when installing the new version so that people who upgrade won't have to change settings and will be given the default behavior upon upgrade

**Pull request merged**